lib/elixir/lib/calendar/iso.ex

```diff
@@ -258,4 +460,24 @@ defmodule Calendar.ISO do
       _ -> :error
     end
   end
+
+  def valid_date?(year, month, day)
+  def valid_date?(year, month, day) when month in (1..12) and day in (1..31) do
+    :calendar.valid_date(year, month, day) and year <= 9999
+  end
+  def valid_date?(year, month, day), do: false
+
+  def valid_time?(hour, minute, second, microsecond)
+  def valid_time?(hour, minute, second, microsecond) when is_integer(microsecond) do
```

Comment:
The contract says this is always a tuple, so I would like to remove this clause, as it will help us catch bugs.